packages/app/src/specs/SpecsList.vue

```diff
@@ -417,6 +420,20 @@ const { refetchFailedCloudData } = useCloudSpecData(
   props.gql.currentProject?.specs as SpecsListFragment[] || [],
 )
 
+const { runAllSpecs, isRunAllSpecsAllowed } = useRunAllSpecs()
+
+function onRunAllSpecs (directory: string) {
+  const filteredSpecs = treeSpecList.value.reduce((acc, node) => {
+    if (node.isLeaf && node.id.startsWith(directory)) {
+      acc.push(node.data?.relative!)
+    }
+
+    return acc
+  }, [] as string[])
+
+  runAllSpecs(filteredSpecs)
+}
```

Comment:
I think it's more idiomatic to do the typing as the first type to `reduce`.

```suggestion
function onRunAllSpecs (directory: string) {
  const filteredSpecs = treeSpecList.value.reduce<string[]>((acc, node) => {
    if (node.isLeaf && node.id.startsWith(directory)) {
      acc.push(node.data?.relative!)
    }

    return acc
  }, [])

  runAllSpecs(filteredSpecs)
}
```

Although I'm guessing this will change anyway with the Spec List refactor + work Ankit is doing on the UI.

Reply:
Done in [b0ba5c2](https://github.com/cypress-io/cypress/pull/24683/commits/b0ba5c2b0e86e79183a9715b58e6fa22c9aa703a)